lib/src/bridge/characteristics_mixin.dart

```diff
@@ -43,11 +49,16 @@ mixin CharacteristicsMixin on FlutterBLE {
           )
           .catchError((errorJson) =>
               Future.error(BleError.fromJson(jsonDecode(errorJson.details))))
-          .then(
-            (rawJsonValue) =>
-                _parseCharacteristicWithValueWithTransactionIdResponse(
-                    peripheral, rawJsonValue),
-          );
+          .then((rawValue) {
+            String rawJsonValue = "";
+            if (rawValue is String) {
+              rawJsonValue = rawValue;
+            }
```

Comment:
This can actually be get rid of by changing to `.invokeMethod<String>(...)`
And yes, it's guaranteed to be string at this point in time.

Reply:
Yes, now I know, I didn't know it then when I wrote this... :/ would have been much more simpler to fixup the bit too loose typing in the code :). But that's how we learn right?

Reply:
> it's guaranteed to be string at this point in time.

As a matter of fact it will have the correct type inferred even of `rawValue` no need to statically denote the type in `then()`-s param list like so:
 - good: `then(rawValue)` inferred to be string cause of invoke's generic type notation
 - bad: `then(String rawValue)` unnecessary and can even break when `invokeMetod<dynamic>`